### ISSUE TITLE
updated module version not recognized; keeps trying to update

### ISSUE THREAD

**Issue by pgnd** [tomhughes/IO-Zlib#7]
after successfully updating v1.13, it continues to try to RE-update the module.

i suspect a missing version string ... ?

```
module_info IO::Zlib
	Name:        IO::Zlib
	Version:     undef
	Directory:   /usr/share/perl5/vendor_perl
	File:        /usr/share/perl5/vendor_perl/IO/Zlib.pm
	Core module: no


cpan -D IO::Zlib
	Loading internal logger. Log::Log4perl recommended for better logging
	Reading '/root/.local/share/.cpan/Metadata'
	  Database was generated on Tue, 10 Jan 2023 18:54:03 GMT
	IO::Zlib
	-------------------------------------------------------------------------
	        (no description)
	        T/TO/TOMHUGHES/IO-Zlib-1.13.tar.gz
	        /usr/share/perl5/vendor_perl/IO/Zlib.pm
	        Installed: 1.11                     <----- !!!!!!!!!
	        CPAN:      1.13  Not up to date     <----- !!!!!!!!!
	        Tom Hughes (TOMHUGHES)
	        tom@compton.nu

cpanm IO::Zlib
	--> Working on IO::Zlib
	Fetching https://cpan.metacpan.org/authors/id/T/TO/TOMHUGHES/IO-Zlib-1.13.tar.gz ... OK
	Fetching https://cpan.metacpan.org/authors/id/T/TO/TOMHUGHES/CHECKSUMS ... OK
	Verifying the SIGNATURE file ... Verified OK
	Configuring IO-Zlib-1.13 ... OK
	Building and testing IO-Zlib-1.13 ... OK                    <----- !!!!!!!!!
	Successfully installed IO-Zlib-1.13 (upgraded from 1.11)    <----- !!!!!!!!!
	1 distribution installed

cpan -D IO::Zlib
	Loading internal logger. Log::Log4perl recommended for better logging
	Reading '/root/.local/share/.cpan/Metadata'
	  Database was generated on Tue, 10 Jan 2023 18:54:03 GMT
	IO::Zlib
	-------------------------------------------------------------------------
	        (no description)
	        T/TO/TOMHUGHES/IO-Zlib-1.13.tar.gz
	        /usr/share/perl5/vendor_perl/IO/Zlib.pm
	        Installed: 1.11                     <----- !!!!!!!!!
	        CPAN:      1.13  Not up to date     <----- !!!!!!!!!
	        Tom Hughes (TOMHUGHES)
	        tom@compton.nu

cpan-outdated -p | cpanm
	--> Working on IO::Zlib
	Fetching https://cpan.metacpan.org/authors/id/T/TO/TOMHUGHES/IO-Zlib-1.13.tar.gz ... OK
	Fetching https://cpan.metacpan.org/authors/id/T/TO/TOMHUGHES/CHECKSUMS ... OK
	Verifying the SIGNATURE file ... Verified OK
	Configuring IO-Zlib-1.13 ... OK
	Building and testing IO-Zlib-1.13 ... OK                    <----- !!!!!!!!!
	Successfully installed IO-Zlib-1.13 (upgraded from 1.11)    <----- !!!!!!!!!
	1 distribution installed
```

I can repeat this without change.

**Comment by tomhughes** [comment]
No idea what that would be I'm afraid. There are no references to 1.11 in the current source.

**Comment by tomhughes** [comment]
Both `META.json` and `META.yml` in the archive say the version is 1.13 and cpan.org seems convinced that is the version so I have no idea why your CPAN client thinks otherwise.

**Comment by pgnd** [comment]
hmm ...

```
module_info IO::Zlib
	Name:        IO::Zlib
	Version:     undef
	Directory:   /usr/share/perl5/vendor_perl
	File:        /usr/share/perl5/vendor_perl/IO/Zlib.pm
	Core module: no

cd /usr/share/perl5/vendor_perl
grep -rlni "1\.11" .
  ./Zlib.pm

grep -i "1\.11" `grep -rlni "1\.11" .`
  our $VERSION = "1.11";
```

digging around a bit ...

**Comment by tomhughes** [comment]
Well the `Zlib.pm` in the archive has 1.13 there:

```
bericote [/tmp/IO-Zlib-1.13] % fgrep 1.13 **/*(.)
ChangeLog:          Change version number to 1.13.
META.json:   "version" : "1.13",
META.yml:version: '1.13'
Zlib.pm:our $VERSION = "1.13";
```

**Comment by tomhughes** [comment]
Anything in `vendor_perl` is an OS packaged version - installing from CPAN will not install there but unless you use `UNINST=1` then CPAN won't cleanup either.

**Comment by tomhughes** [comment]
I'm guessing you have the `perl-IO-Zlib` rpm installed and that is where 1.11 is coming from.

The CPAN install will either have gone to `/usr/share/perl5` or `/usr/local/lib64/perl5/5.36` but I'm not sure which. Normally it would be the latter (the site perl directory) but I think now it's part of perl core it might be the former.

**Comment by tomhughes** [comment]
In fact almost certainly that is what has happened as the Fedora perl search path is:

```
  @INC:
    /usr/local/lib64/perl5/5.36
    /usr/local/share/perl5/5.36
    /usr/lib64/perl5/vendor_perl
    /usr/share/perl5/vendor_perl
    /usr/lib64/perl5
    /usr/share/perl5
```

so your rpm installed 1.11 will take priority over the CPAN installed 1.13 in `/usr/share/perl5`.

Which is why you shouldn't use CPAN on rpm managed systems, and especially not to install new versions of modules which are in perl core.

**Comment by pgnd** [comment]
@tomhughes 

since you closed, up to you, but fyi:

```
I raised this issue in #irc

[15:17] <pgnd> hi.  i keep staring at this syntax :-/
[15:17] <pgnd> does this :  INSTALLDIRS  => $] >= 5.00903 ? "perl" : "site",
[15:17] <pgnd> in fact read "If perl version is >= v5.93, then install in 'perl' path , if not the 'site' path?
[15:18] <LeoNerd> No, version v5.9.3
[15:18] <pgnd> LeoNerd: oops. yeah.  but if true, then into 'perl', right?
[15:19] <LeoNerd> indeed
[15:19] <pgnd> ty o/
[15:19] <LeoNerd> It's a workaround for an old thing we fixed a very long time ago... at that version in fact
[15:19] <thrig> 2006 calling
[15:20] <pgnd> Noted.  It's in a module that's misbehaving (I think) on cpanm install, ending up lower in @INC hierarchy than vendor/perl dirs.
[15:22] <Grinnz> actually that's the version it was added in core; that bug was fixed in 5.012
[15:22] <Grinnz> what module are you looking at?
[15:22] <Grinnz> all of those that fucked that up should have been fixed by now
[15:24] <pgnd> Grinnz: IO::ZLib.  My issue was this, https://github.com/tomhughes/IO-Zlib/issues/7 , but closed without further discussion before I could ask.  Then I followed the Extutils::Makemaker rabbit hole and stumbled on that INSTALLDIRS stanza, which seemed ... odd to me.
[15:24] <Paperbot> Link title(s): [ updated module version not recognized; keeps trying to update · Issue #7 · tomhughes/IO-Zlib · Gi... ]
[15:25] <Grinnz> perlbot: corelist IO::Zlib
[15:25] <+perlbot> Grinnz: IO::Zlib Added to perl core as of 5.009003
[15:25] <Grinnz> yeah, that's broken, it should have a maximum version
[15:26] <pgnd> Best to take it up on github with module author?  or since it's in perl 'core' ... somehwere else?
```
